docs/developer/decision-records/2023-02-07-command-handler-result/README.md

```diff
@@ -0,0 +1,46 @@
+# Title of the Decision Record
+
+## Decision
+
+Return the `CommandHandler`s' thrown exceptions,that are catched by the `CommandRunner` as a `Result`.  
+
+
+## Rationale
+
+
+At the moment, some `CommandHandler`s throw exceptions to notify validation errors, that are catched by the `CommandRunner`, but these errors are seen as unexpected.
+In order to make these errors detection easier, the `CommandHandler.handle` is modified to return a `Result` that is evaluated by the `CommandRunner` , then the `Result` would be returned to the `CommandProcessor` and logged accordingly.
+
+## Approach
+
+
+1. The `CommandHandler.handle` return a `Result`:
+```java
+public interface CommandHandler<T extends Command> {
+
+    Class<T> getType();
+    
+    Result<Void> handle(T command);
+}
+```
+2. Make the `CommandRunner.runCommand` evaluates the `Result` that is returned from the `CommandHandler`:
+```java
+
+
+   public <T extends C> Result<Void> runCommand(T command) {
+
+        @SuppressWarnings("unchecked") var commandClass = (Class<T>) command.getClass();
+
+        var handler = commandHandlerRegistry.get(commandClass);
+        if (handler == null) {
+            command.increaseErrorCount();
+            return Result.failure("No command handler found for command type " + commandClass);
+        }
+        try {
+            var result = handler.handle(command);
+            return result;
+        }
+        .....
+   }
+
+```
```

Comment:
I'd add that the `CommandHandler`s that are currently throwing exceptions explicitly should return a failed `Result` instead.

Reply:
Additionally I would also suggest, that the indentation could be fixed, but it's just a smaller syntax mistake. Nothing which blocks in my opinion the merge.